tests/test_metadata_serialization.py

```diff
@@ -48,8 +50,61 @@ def wrapper(test_cls: "TestSerialization"):
     return real_decorator
 
 
+class TestSigned(Signed):
+    """Used for testing the abstract "Signed" class."""
+
+    _signed_type = "signed"
+
+    def __init__(
+        self,
+        version: int,
+        spec_version: str,
+        expires: datetime,
+        unrecognized_fields: Optional[Mapping[str, Any]]
+    ) -> None:
+        super().__init__(
+            version, spec_version, expires, unrecognized_fields
+        )
+
+    @classmethod
+    def from_dict(cls, signed_dict: Dict[str, Any]) -> "Signed":
+        common_args = super()._common_fields_from_dict(signed_dict)
+        # All fields left in the signed_dict are unrecognized.
+        return cls(*common_args, signed_dict)
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Returns the dict representation of self."""
+        return super()._common_fields_to_dict()
+
+
 class TestSerialization(unittest.TestCase):
 
+    invalid_signed: DataSet = {
```

Comment:
there's a lot of test cases here that are unlikely to be useful (e.g. 5 invalid spec versions  is just overkill) but since the tests are so fast... I guess no harm done?

Reply:
I removed one of the cases one digit spec_version, the others seem relevant to the checks we are making.

Reply:
ok, but do keep in mind that the idea isn't to test the lines of code that we happen to have in there right now: the idea is to have test data that covers as much of the possible inputs as practically useful -- since we can't cover 100% of the possibilities we inevitably end up making decisions on what's important: we test the inputs we think could be interesting.